### PR TITLE
@damassi => [WIP] Windowed pagination in filter component

### PR DESCRIPTION
**Pull request by mzikherman**
This is...v v unstyled. It's just to get/exercise the functionality + Metaphysics adjustments.

Screencap demonstrating prev/next and page skipping.

bp

This wound up being very straightforward, using https://github.com/artsy/metaphysics/pull/1086 (I'll follow up on that PR).

### CODE REVIEW

File: src/Components/ArtistFilter/Artworks.tsx [hunk 1]
```diff
@@ -14,41 +9,91 @@ import { Artworks_filtered_artworks } from "../../__generated__/Artworks_filtere
...
+    }
+  }
+
+  loadPrevPage(filters) {
+    if (filters.state.page > 1) {
+      return this.loadPage(filters, filters.state.page - 1)
+    }
+  }
+
+  renderPaginationBar(filters) {
```
Comment (mzikherman) [inline src/Components/ArtistFilter/Artworks.tsx]: This just renders all the numbers from 1 up until `last page`, and lets you click to select any particular one. Basically just to demonstrate that functionality, the real pagination bar component will have a bit of logic as to which pages to display (1... 6, 7, 8 .... 19) kinda thing.

File: src/Components/ArtistFilter/Artworks.tsx [hunk 2]
```diff
@@ -14,41 +9,91 @@ import { Artworks_filtered_artworks } from "../../__generated__/Artworks_filtere
...
+    let page
+    const pages = []
+    for (
+      page = 1;
+      page < this.props.filtered_artworks.artworks.totalPages + 1;
+      page++
+    ) {
+      pages.push(
+        <span
+          data-page={page}
```
Comment (mzikherman) [inline src/Components/ArtistFilter/Artworks.tsx]: Oh yea, we can definitely tackle this when we make this a more proper component, but I couldn't figure out how to get the click handler to work on `page`, b/c of references all the click handlers just point to the last page (probably something simple, can figure out later when we take another pass).

File: src/Components/ArtistFilter/Filter.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import ArtworksContent from "./Artworks"
 
 import { Filter_artist } from "../../__generated__/Filter_artist.graphql"
 
-interface Props {
+export interface Props {
```
Comment (mzikherman) [inline src/Components/ArtistFilter/Filter.tsx]: This was b/c of a TS lint error, we are exporting a class which wraps this [private] interface.

File: src/Components/ArtistFilter/Artworks.tsx (hunk 2 above)
Comment (damassi) [reply src/Components/ArtistFilter/Artworks.tsx]: npnp, we can clean up in another pass

**Comment by damassi** [comment]
@mzikherman - this still mergable or should we wait for the MP one above? Awesome that this is coming together so simply, very little logic in here 👍

**Comment by mzikherman** [comment]
@damassi updated the PR title to have "WIP". This (and the MP PR) approach is to add `page/size` to connections.

I'll leave those open and add another pair of PRs (MP and Reaction) that keeps the existing cursor-based args/pattern, but adds a 'table of contents' to the metadata so clients can figure out the cursors for each page. There was discussion in Slack that this might be preferable as it's an extension of the connection spec, rather than a deviation.

**Comment by mzikherman** [comment]
I'll close this since it seems we won't be doing with `page/size` here, but with the cursor info approach in https://github.com/artsy/reaction/pull/746/files

**Pull request closed without merging**